app/controllers/api/automated_deploys_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+# deploys that automatically get triggered when new hosts come up or are restarted
+class Api::AutomatedDeploysController < Api::BaseController
+  STAGE_NAME = 'Automated Deploys'
+
+  before_action :find_or_create_stage
+  before_action :find_deploy_group
+  before_action :find_last_deploy
+
+  def create
+    deploy_service = DeployService.new(current_user)
+    deploy = deploy_service.deploy!(
+      @stage,
+      reference: @last_deploy.reference,
+      buddy_id: @last_deploy.buddy_id,
```

Comment:
We should check with compliance on whether these deploys should be tagged as having a buddy. Since nobody actually approved this particular deploy, I think that column should be blank.

Though it would be nice if this deploy had a reference to the `@last_deploy` that it copied from, or otherwise tag this as an auto-deploy.